### PR TITLE
[RF] Updates for templateflow

### PR DESCRIPTION
**Pull request by oesteban**
To work with https://github.com/poldracklab/fmriprep/pull/1334

### CODE REVIEW

**Comment by oesteban** [comment]
Any crazy stuff that I don't see?

File: niworkflows/data/__init__.py [hunk 1]
```diff
@@ -8,20 +8,8 @@
...
-    get_ants_oasis_template,
-    get_ants_oasis_template_ras,
-    get_mni_epi,
-    get_mni152_nlin_sym_las,
-    get_mni152_nlin_sym_ras,
-    get_mni_icbm152_linear,
-    get_mni_icbm152_nlin_asym_09c,
-    get_oasis_dkt31_mni152,
-    get_hcp32k_files,
-    get_conte69_mesh,
```
Comment (effigies) [inline niworkflows/data/__init__.py]: Should we go through a deprecation cycle for these? Or is niworkflows more of a makes-no-promises sort of library?

File: niworkflows/interfaces/masks.py [hunk 2]
```diff
@@ -259,20 +262,50 @@ def _generate_report(self):
...
+
+        nsegs = len(seg_files)
+        if nsegs == 1:
+            if not levels:  # Segmentation
+                levels = np.unique(np.round(
+                    nb.load(seg_files[0]).get_data()).astype(int))
+                levels = (levels[levels > 0] - 0.5).tolist()
+            nsegs = len(levels)
+        else:
+            if not levels:
```
Comment (effigies) [inline niworkflows/interfaces/masks.py]: `elif not levels:`?

File: niworkflows/interfaces/masks.py [hunk 3]
```diff
@@ -259,20 +262,50 @@ def _generate_report(self):
...
+            colors = colors + color_palette("husl", missing)
+
+        sortedc = []
+        start = 0
+        for l in levels:
+            end = start + len(l)
+            sortedc.append(colors[start:end])
+            start = end
 
+        print(levels, sortedc)
```
Comment (effigies) [inline niworkflows/interfaces/masks.py]: Suspect this shouldn't be here.

File: niworkflows/viz/utils.py [hunk 4]
```diff
@@ -283,28 +283,32 @@ def plot_segs(image_nii, seg_niis, out_file, bbox_nii=None, masked=False,
...
 def _plot_anat_with_contours(image, segs=None, compress='auto',
                              **plot_params):
-    assert segs is not None
-
+    nsegs = len(segs or [])
     plot_params = plot_params or {}
-    colors = plot_params.pop('colors', []) or []  # colors should not be None
-    nsegs = len(segs)
+    colors = plot_params.pop('colors', None) or []
+    levels = plot_params.pop('levels', None) or []
```
Comment (effigies) [inline niworkflows/viz/utils.py]: Suspect I'm missing why you changed it, but isn't this equivalent to:

```Python
colors = plot_params.pop('colors', [])
levels = plot_params.pop('levels', [])
```

File: niworkflows/data/__init__.py (hunk 1 above)
Comment (oesteban) [reply niworkflows/data/__init__.py]: Given that this was a pretty unscalable approach and that fmriprep is the only client of niworkflows... I'd go with makes-no-promises. That will also push people to update fmriprep if they are not using containers, which is positive in the long term.

We should consider always setting a maximum version for niworkflows in fmriprep, though.

File: niworkflows/viz/utils.py (hunk 4 above)
Comment (oesteban) [reply niworkflows/viz/utils.py]: Unfortunately not:
```Python
>>> plot_params = {'colors': None, 'levels': None}
>>> colors = plot_params.pop('colors', [])
... levels = plot_params.pop('levels', [])
>>> colors is None
True
>>> levels is None
True
```

Many times they are set and `None`, so I had to change to that ugly `or`. Maybe there's a better way of doing this I'm missing (e.g. cleaning the dictionary of `None`s before, however I'm not sure other kw may use None as a valid value or maybe replacing and then pop... I don't have a preference).

Comment (effigies) [reply niworkflows/viz/utils.py]: Got it. Maybe just a quick comment (e.g. "None is a valid parameter") so the next person to look at this won't think the same thing.

**Comment by effigies** [comment]
Sounds good.

**Comment by effigies** [comment]
Good to go? Looks like poldracklab/fmriprep#1334 is passing.

**Comment by effigies** [comment]
(Or do we want to wait for the next release, to avoid too much churn?)

**Comment by oesteban** [comment]
I'm fixing the colors of `ROIsPlot`. If I'm not done by the end of the day, let's push it to the next release.

**Comment by oesteban** [comment]
Alright, this is finished. I've tested locally and the plots look the same as before the PR 👍.

Currently testing fmriprep here: https://circleci.com/workflow-run/d3fa2769-7013-4fce-83f0-23e9a32ae158

Right after merging this, I'll release 0.5.0 (I think the big change in templates deserves that minor bump)

**Pull request merged**